Adrien/global docs text search parallel

why-not-try-calmer opened the pull request:
My bad, I had forgotten that I had that old branch under PR (#46 ) when I was working on this one all the while.

why-not-try-calmer committed:
reset

- readme

Update defrag/modules/docs.py

Co-authored-by: Jens <karatekhd@opensuse.org>

Update defrag/modules/docs.py

Co-authored-by: Jens <karatekhd@opensuse.org>

Update defrag/modules/docs.py

Co-authored-by: Jens <karatekhd@opensuse.org>

- removed unnecessary line
- adjusted tests

- small optimization

- removed unnecessary call

- added parallel computation for building and searching indexes on a fresh build

about to rebase

why-not-try-calmer committed:
- typo

why-not-try-calmer committed:
-

KaratekHD commented on defrag/modules/docs.py:
```diff
@@ -0,0 +1,159 @@
+import sys
+from typing import Any, Dict, List, Tuple
+
+__MOD_NAME__ = "docs"
```
```suggestion
__MOD_NAME__ = "documentation"
```

On /docs, FastAPI serves autogenerated documentation

KaratekHD commented on defrag/modules/docs.py:
```diff
@@ -0,0 +1,159 @@
+        return leap_worker.result() + tw_worker.result()
+
+
+@app.get("/" + __MOD_NAME__ + "/single/{source}/{keywords}")
```
```suggestion
@app.get("/" + __MOD_NAME__ + "/single/{source}/")
```
This would make it so that we take the search term as a query parameter, which I think is nicer.

KaratekHD commented on defrag/modules/docs.py:
```diff
@@ -0,0 +1,159 @@
+    return QueryResponse(query=Query(service="search_docs"), results_count=len(results), results=results)
+
+
+@app.get("/" + __MOD_NAME__ + "/merged/{keywords}")
```
```suggestion
@app.get("/" + __MOD_NAME__ + "/merged/")
```
This would make it so that we take the search term as a query parameter, which I think is nicer.

KaratekHD commented on requirements.txt:
```diff
@@ -11,16 +11,13 @@ click==8.0.1
+PyYAML==5.4.1
 redis==3.5.3
 starlette==0.14.2
 typing-extensions==3.10.0.0
```
```suggestion
typing-extensions==3.10.0.0
lunr==0.6.0
```

why-not-try-calmer committed:
Update defrag/modules/docs.py

good catch

Co-authored-by: Jens <karatekhd@opensuse.org>

KaratekHD reviewed (changes requested):
Sorry, hit the wrong button, meant to request changes

why-not-try-calmer committed:
Update requirements.txt

np

Co-authored-by: Jens <karatekhd@opensuse.org>

why-not-try-calmer committed:
Update defrag/modules/docs.py

:D you are correct, I rushed through these FastAPI docs and I perhaps need to check them again

Co-authored-by: Jens <karatekhd@opensuse.org>

why-not-try-calmer committed:
Update defrag/modules/docs.py

Co-authored-by: Jens <karatekhd@opensuse.org>

KaratekHD reviewed (approved):
Ready to merge this if you're ready

KaratekHD committed:
Fix docs tests
